service/history/api/create_workflow_util.go

```diff
@@ -75,6 +78,9 @@ func NewWorkflowWithSignal(
 		workflowID,
 		runID,
 	)
+	newMutableState.GetExecutionInfo().VersioningInfo = &workflowpb.WorkflowExecutionVersioningInfo{
+		VersioningOverride: startRequest.GetVersioningOverride(),
+	}
```

Comment:
Todo: need to update build id search attributes somewhere near here + add a reachability test about this